cfme/cloud/keypairs.py

```diff
@@ -199,6 +199,24 @@ def refresh():
         view.entities.flash.assert_no_error()
         view.entities.flash.assert_success_message(flash_message)
 
+    def add_tag(self, tag, **kwargs):
+        """Tags the Keypair by given tag"""
+        navigate_to(self, 'Details')
+        mixins.add_tag(tag, **kwargs)
+
+    def remove_tag(self, tag, **kwargs):
+        """Untag the Keypair by given tag"""
+        navigate_to(self, 'Details')
+        mixins.remove_tag(tag)
+
+    def exists(self):
+        try:
+            view = navigate_to(self, 'Details')
+            view.is_displayed()
+            return True
+        except Exception:
```

Comment:
Let's avoid diaper pattern here, the navigation raises KeyPairNotFound exception if the row isn't found.